lib/org/chromium/webidl/Parser.js

```diff
@@ -0,0 +1,832 @@
+    'org.chromium.webidl.ast.MapLike',
+    'org.chromium.webidl.ast.Member',
+    'org.chromium.webidl.ast.MemberData',
+    'org.chromium.webidl.ast.Namespace',
+    'org.chromium.webidl.ast.NonUnionType',
+    'org.chromium.webidl.ast.OctInteger',
+    'org.chromium.webidl.ast.Operation',
+    'org.chromium.webidl.ast.OperationQualifier',
+    'org.chromium.webidl.ast.ParameterizedType',
+    'org.chromium.webidl.ast.Partial',
+    'org.chromium.webidl.ast.Serializer',
+    'org.chromium.webidl.ast.SerializerPattern',
+    'org.chromium.webidl.ast.SerializerPatternType',
+    'org.chromium.webidl.ast.SetLike',
+    'org.chromium.webidl.ast.SimpleType',
+    'org.chromium.webidl.ast.StaticMember',
+    'org.chromium.webidl.ast.String',
+    'org.chromium.webidl.ast.Stringifier',
+    'org.chromium.webidl.ast.Type',
+    'org.chromium.webidl.ast.TypeSuffix',
+    'org.chromium.webidl.ast.Typedef',
+    'org.chromium.webidl.ast.UnionType',
+    'org.chromium.webidl.parsers.TokenParsers',
+  ],
+
+  properties: [
+    {
+      // The core query parser. Needs a fieldname symbol added to function
+      // properly.
+      name: 'symbolsFactory',
+      value: function(
+        sym, literal, range, str, repeat, plus, optional, alt, seq, seq1,
+        notChars, trepeat, tseq1, tplus, tseq
+      ) {
+        return {
+          // Common patterns.
+          _09: range('0', '9'),
+          r09: str(repeat(sym('_09'))),
+          p09: str(plus(sym('_09'))),
+          om: optional('-'),
+          opm: optional(alt('+', '-')),
+          Ee: alt('E', 'e'),
+          AZ: range('A', 'Z'),
+          az: range('a', 'z'),
+
+          // Common tokens.
+          integer: seq(
+            sym('om'),
+            alt(
+              sym('dec'),
+              sym('hex'),
+              sym('oct')
+            )
+          ),
+          dec: str(seq(range('1', '9'), sym('r09'))),
+          hex: str(seq('0', alt('X', 'x'), str(plus(alt(
+            range('0', '9'), range('A', 'F'), range('a', 'f')
+          ))))),
+          oct: str(seq('0', str(repeat(range('0', '7'))))),
+          float: seq(
+            sym('om'),
+            alt(
+              str(seq(
+                alt(
+                  str(seq(sym('p09'), '.', sym('r09'))),
+                  str(seq('.', sym('p09')))
+                ),
+                optional(
+                  str(seq(sym('Ee'), sym('opm'), sym('p09')))
+                )
+              )),
+              str(seq(sym('p09'), sym('Ee'), sym('opm'), sym('p09')))
+            )
+          ),
+
+          // TODO: This has been relaxed to parse identifiers like
+          // "__content" in Gecko's IDL. It should be
+          // /_?[A-Za-z][0-9A-Z_a-z-]*/.
+          identifier: seq(
+            alt(sym('AZ'), sym('az'), '_'),
+            str(repeat(alt(sym('AZ'), sym('az'), sym('_09'), '_', '-')))
+          ),
+          string: seq1(1, '"', str(repeat(notChars('"'))), '"'),
```

Comment:
Pshhh. :wave:

Yeah. You're right. Haven't encountered any examples, but they may eventually turn up.